File: java/src/test/java/io/binac/leetcode/DistinctSubsequencesTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class DistinctSubsequencesTest {
    private static final DistinctSubsequences.Solution1 SOLUTION1 = new DistinctSubsequences.Solution1();

    private final String S1 = "rabbbit", T1 = "rabbit";
    private final int EXPECTED1 = 3;

    private final String S2 = "babgbag", T2 = "bag";
    private final int EXPECTED2 = 5;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.numDistinct(S1, T1));
        assertEquals(EXPECTED2, SOLUTION1.numDistinct(S2, T2));
    }
}

File: cpp/tests/DistinctSubsequencesTest.cpp
#include "DistinctSubsequences.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(DistinctSubsequences, Solution115_1) {
  auto S115_1 = Solution115_1();

  EXPECT_EQ(3, S115_1.numDistinct("rabbbit", "rabbit"));
  EXPECT_EQ(5, S115_1.numDistinct("babgbag", "bag"));
}